parseclj.el

```diff
@@ -30,9 +30,53 @@
 
 ;;; Code:
 
+(require 'map)
 (require 'parseclj-parser)
 (require 'parseclj-ast)
 
+(defun parseclj-alist (&rest kvs)
+  "Create an association list from the given keys and values KVS.
+Arguments are simply provided in sequence, rather than as lists or cons cells.
+For example: (a-alist :foo 123 :bar 456)"
+  (mapcar (lambda (kv) (cons (car kv) (cadr kv))) (seq-partition kvs 2)))
+
+(defun parseclj-hash-table (&rest kvs)
+  "Create a hash table from the given keys and values KVS.
+Arguments are simply provided in sequence, rather than as lists
+or cons cells. As \"test\" for the hash table, equal is used. The
+hash table is created without extra storage space, so with a size
+equal to amount of key-value pairs, since it is assumed to be
+treated as immutable.
+For example: (a-hash-table :foo 123 :bar 456)"
+  (let* ((kv-pairs (seq-partition kvs 2))
+         (hash-map (make-hash-table :test 'equal :size (length kv-pairs))))
+    (seq-do (lambda (pair)
+              (puthash (car pair) (cadr pair) hash-map))
+            kv-pairs)
+    hash-map))
+
+(defun parseclj-alist-assoc (coll k v)
+  "Like parseclj-alist-assoc but actually works as advertised, not
+introducing duplicate keys"
+  (if (map-contains-key coll k)
+      (mapcar (lambda (entry)
+                (if (equal (car entry) k)
+                    (cons k v)
```

Comment:
Not something I thought about much since the whole point of value semantics is that these would be indistinguishable, but in case they are not I do think it's more useful for the newer `k` to replace the old.